src/irc/core/irc.c

```diff
@@ -382,14 +406,14 @@ static char *irc_parse_prefix(char *line, char **nick, char **address)
 /* Parse command line sent by server */
 static void irc_parse_incoming_line(IRC_SERVER_REC *server, char *line)
 {
-	char *nick, *address;
+	char *nick, *address, *tags;
 
 	g_return_if_fail(server != NULL);
 	g_return_if_fail(line != NULL);
 
-	line = irc_parse_prefix(line, &nick, &address);
-	if (*line != '\0')
-		signal_emit_id(signal_server_event, 4, server, line, nick, address);
+	line = irc_parse_prefix(line, &nick, &address, &tags);
+	if (*line != '\0' || tags != NULL)
```

Comment:
So wait, this is handling tags and no line like...

    @tags \r\n

I don't think that's a valid irc message (the command is always required) or something that makes sense to have special cases for.

Reply:
wouldn't it be better to preserve and make accessible such an invalid line nevertheless?

Reply:
Yeah, I don't know. I guess it doesn't really make other things more complicated and it's harmless to do so.